src/Core/Checkout/Cart/Order/RecalculationService.php

```diff
@@ -360,9 +360,10 @@ private function recalculateCart(Cart $cart, SalesChannelContext $context): Cart
             $cart = $this->processor->process($cart, $live, $behavior);
 
             // validate cart against the context rules
-            $validated = $this->cartRuleLoader->loadByCart($live, $cart, $behavior);
+            $validatedCart = $this->cartRuleLoader->loadByCart($live, $cart, $behavior)->getCart();
+            $validatedCart->setErrors($cart->getErrors());
```

Comment:
Errors should be merged so new ones don't get lost:
```suggestion
            $validatedCart->addErrors(...$cart->getErrors());
```